contracts/hardhat.config.ts

```diff
@@ -46,12 +46,18 @@ export default {
     target: 'ethers-v5',
   },
   networks: {
+    network: {
+      url: process.env.NODE_HTTP_URL || '',
+    },
     hardhat: {
       allowUnlimitedContractSize: Boolean(
         process.env.ALLOW_UNLIMITED_CONTRACT_SIZE,
       ),
     },
   },
+  etherscan: {
+    apiKey: process.env.ETHERSCAN_API_KEY,
+  },
```

Comment:
Yes! I used to have exactly what you suggested:
```
 etherscan: {
    [process.env.NETWORK_NAME]: process.env.EXPLORER_API_KEY,
  },
```
with this dynamic name, we have to 1) let users specify the NETWORK_NAME in .dev, which is yet another variable 2) Since this is a dynamic name(not value), when people run commands like "pnpm clean", they have to do "NETWORK_NAME=mainnet pnpm clean". Not sure how much burden this is, so I explored and changed it to simply the following:

```
etherscan: {
    apiKey: process.env.ETHERSCAN_API_KEY,
  },
```
But as you can tell, if we expand to non-evm chains, we have to change to per-network basis. Given that we support only ethereum chains in this PR, the current design has minimum disruption to users, but I am open to suggestions. WDYT?

Reply:
I would dynamically generate the config so that commands don't error if no env var provided. something like this...
```ts
```